gneiss/plot/heatmap.py

```diff
@@ -0,0 +1,124 @@
+"""
+Plotting functions (:mod:`gneiss.plot`)
+===============================================
+
+.. currentmodule:: gneiss.plot
+
+This module contains plotting functionality
+
+Functions
+---------
+
+.. autosummary::
+   :toctree: generated/
+
+   heatmap
+
+"""
+
+import numpy as np
+import pandas as pd
+from skbio.stats.composition import clr_inv
+from collections import OrderedDict
+from ete3 import Tree, TreeStyle, AttrFace, ProfileFace
+from ete3 import ClusterNode
+from ete3.treeview.faces import add_face_to_node
+from gneiss.layouts import default_layout
+from gneiss.balances import _attach_balances
+import io
+
+
+def heatmap(table, tree, layout=None, cmap='viridis', **kwargs):
+    """ Plots tree on heatmap
+
+    Parameters
+    ----------
+    table : pd.DataFrame
+        Contingency table where samples correspond to rows and
+        features correspond to columns.
+    tree : skbio.TreeNode
+        A strictly bifurcating tree defining a hierarchical relationship
+        between all of the features within `table`.
+    layout : function, optional
+        A layout for formatting the tree visualization. Must take a
+        `ete.tree` as a parameter.
+    cmap: matplotlib colormap
+        String or function encoding matplotlib colormap.
+    rowlabel_size : int
+        Size of row labels
+    width : int
+        Heatmap cell width
+    height : int
+        Heatmap cell height
+
+    Returns
+    -------
+    ete.Tree
+        ETE tree object that will be plotted.
+    ete.TreeStyle
+        ETE TreeStyle that decorates the tree and heatmap visualization.
+    """
+
+    params = {'rowlabel_size': 8, 'width':200, 'height':14, 'cmap':'viridis'}
+    for key in params.keys():
+        params[key] = kwargs.get(key, params[key])
+    fsize = params['rowlabel_size']
```

Comment:
Why creating four new variables instead of using `params[x]` directly? (I am just curious. I have no problem with it.)